src/components/FlowChart/ChartNode.tsx

```diff
@@ -1,50 +1,87 @@
+import Pod from "./Pod"
+import { showModal } from "../../redux/global/global.actions"
+import { useDispatch, useSelector } from "react-redux"
+import { selectSelectedFlow } from "../../redux/flows/flows.selectors"
+import styled from "@emotion/styled"
 
-type Props = {
-  node: {
-    properties: any;
-    label: string | undefined;
-    id?: string;
-  };
-};
+export const ChartNodeElement = styled.div`
+  min-width: 16rem;
+  cursor: move;
+  text-align: center;
+  font-size: 14px;
+  background: #fff;
+  font-weight: 500;
+  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
+  border-radius: 0.25em;
+  transition: 0.2s;
+  border: 1px solid rgba(0, 153, 153, 0.3);
+`
 
-export default function ChartNode({ node }: Props) {
-  const { properties, label } = node;
-  const list: any = [];
-  const dispatch = useDispatch();
-  Object.keys(properties).forEach((prop, idx) => {
-    if (properties[prop] && prop !== "name")
-      list.push(
-        <div key={idx}>
-          <span className="text-bold mr-1">{prop}:</span>
-          {properties[prop]}
-        </div>
-      );
-  });
-  const isSpecial = Object.keys(properties).length > 0;
-  let labelText = typeof label === "undefined" ? properties.name : label || "";
-  const type = useSelector(selectFlowChart).type;
-  return (
-    <div
-      className={`chart-node`}
-      id={`chart-node-${label}`}
-      onDoubleClick={() => {
-        type === "user-generated" &&
-          dispatch(showModal("podEdit", { nodeId: node.id }));
-      }}
-    >
-      <div className="node-header">
-        <div className={`p-1 ${isSpecial ? "special" : ""}`}>
-          <p className="m-1">
-            <span className="text-bold">
-              {labelText || <span className="text-warning">Empty Pod</span>}
-            </span>
-          </p>
-        </div>
-      </div>
-    </div>
-  );
+const NodePortTop = styled(Handle)`
+  margin-top: -0.2rem;
+  background-color: white;
+  width: 1rem;
+  height: 1rem;
+  border-radius: 0.5rem;
+  border: 1px solid rgba(0, 153, 153, 0.3); //todo use theming
+  display: flex;
+  align-items: center;
+  justify-content: center;
+
+  &:after {
+    display: block;
+
+    content: "";
+    background-color: #099;
+    color: white;
+    border-radius: 0.25rem;
+    width: 0.5rem;
+    height: 0.5rem;
+  }
+`
+
+const NodePortBottom = styled(NodePortTop)`
+  bottom: -0.45rem;
+`
+
+type NodeType = "Pod" | "Gateway"
+export default function ChartNode(type: NodeType) {
+  switch (type) {
+    case "Pod":
+      return function ChartNode({ id, data }: Node) {
+        const type = useSelector(selectSelectedFlow).type
+        const dispatch = useDispatch()
+        return (
+          <ChartNodeElement
+            onDoubleClick={() => {
+              type === "user-generated" &&
+                dispatch(showModal("podEdit", { nodeId: id }))
+            }}
+          >
+            <NodePortTop type="target" position={Position.Top} />
```

Comment:
Most of what needs to be rendered for `Pod` and `Gateway` looks similar except for `<NodePortTop>`. If this is the only difference, how about making only this part conditional?